test/models/Reads/PensionSchemeAdministratorReadsSpec.scala

```diff
@@ -0,0 +1,129 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.Reads
+
+import models.{Reads => _, _}
+import org.scalatest.{MustMatchers, OptionValues, WordSpec}
+import play.api.libs.functional.syntax._
+import play.api.libs.json.Reads._
+import play.api.libs.json.{Json, _}
+
+class PensionSchemeAdministratorReadsSpec extends WordSpec with MustMatchers with OptionValues with Samples {
+  "JSON Payload of a PSA" should {
+    "Map to a valid PensionSchemeAdministrator object" when {
+      val input = Json.obj(
+        "legalStatus" -> "Individual",
+        "sapNumber" -> "NumberTest",
+        "noIdentifier" -> JsBoolean(true),
+        "customerType" -> "TestCustomer",
+        "contactDetails" -> Json.obj("phone" -> "07592113", "email" -> "test@test.com"),
+        "companyAddressYears" -> JsString("over_a_year"),
+        "companyAddressId" -> JsObject(Map("addressLine1" -> JsString("line1"),"addressLine2" -> JsString("line2"),"addressLine3" -> JsString("line3"),"addressLine4" -> JsString("line4"),"postalCode" -> JsString("NE1"),"countryCode" -> JsString("GB"))))
+
+      "We have a valid legalStatus" in {
+        val result = Json.fromJson[PensionSchemeAdministrator](input)(apiReads).asOpt.value
+
+        result.legalStatus mustEqual pensionSchemeAdministratorSample.legalStatus
+      }
+
+      "We have a valid sapNumber" in {
+        val result = Json.fromJson[PensionSchemeAdministrator](input)(apiReads).asOpt.value
+
+        result.sapNumber mustEqual pensionSchemeAdministratorSample.sapNumber
+      }
+
+      "We have a valid noIdentifier" in {
+        val result = Json.fromJson[PensionSchemeAdministrator](input)(apiReads).asOpt.value
+
+        result.noIdentifier mustEqual pensionSchemeAdministratorSample.noIdentifier
+      }
+
+      "We have valid customerType" in {
+        val result = Json.fromJson[PensionSchemeAdministrator](input)(apiReads).asOpt.value
+
+        result.customerType mustEqual pensionSchemeAdministratorSample.customerType
+      }
+
+      "We have a valid idType" in {
+        val result = Json.fromJson[PensionSchemeAdministrator](input + ("idType" -> JsString("TestId")))(apiReads).asOpt.value
+
+        result.idType mustEqual pensionSchemeAdministratorSample.idType
+      }
+
+      "We have a valid idNumber" in {
+        val result = Json.fromJson[PensionSchemeAdministrator](input + ("idNumber" -> JsString("TestIdNumber")))(apiReads).asOpt.value
+
+        result.idNumber mustEqual pensionSchemeAdministratorSample.idNumber
+      }
+
+      "We have a moreThanTenDirectors flag" in {
+        val result = Json.fromJson[PensionSchemeAdministrator](input + ("moreThanTenDirectors" -> JsBoolean(true)))(apiReads).asOpt.value
+
+        result.numberOfDirectorOrPartners.value.isMorethanTenDirectors mustEqual pensionSchemeAdministratorSample.numberOfDirectorOrPartners.value.isMorethanTenDirectors
+      }
+
```

Comment:
Is a false flag implied if field is not set?

Reply:
Yes, this flag is not set if we don't have more than ten directors. The object itself is not required in the schema so I'm assuming that having it not set means it is false (from DES point of view)